task/task.go

```diff
@@ -120,3 +121,115 @@ func (t *TaskWriteTests) Run(repository *repository.Repository) (repositoryAsses
 }
+
+// TaskCodeRepair holds the code repair task.
+type TaskCodeRepair struct {
+	// ResultPath holds the directory path where results should be written to.
+	ResultPath string
+
+	// Language holds the language for which the task should be evaluated.
+	Language language.Language
+	// Model holds the model which the task should be evaluated.
+	Model evalmodel.Model
+
+	// Logger holds the logger for this tasks.
+	Logger *log.Logger
+}
+
+var _ Task = (*TaskCodeRepair)(nil)
+
+// NewCodeRepairTask returns a code repair task.
+func NewCodeRepairTask(logger *log.Logger, resultPath string, model evalmodel.Model, language language.Language) (task Task) {
+	return &TaskCodeRepair{
+		ResultPath: resultPath,
+		Language:   language,
+		Model:      model,
+		Logger:     logger,
+	}
+}
+
+// Identifier returns the code repair task identifier.
+func (t *TaskCodeRepair) Identifier() taskcontext.Identifier {
+	return taskcontext.IdentifierCodeRepair
+}
+
+// Run performs source code repairing in a repository with compilation errors.
+func (t *TaskCodeRepair) Run(repository *repository.Repository) (repositoryAssessment metrics.Assessments, problems []error, err error) {
+	log, logClose, err := log.WithFile(t.Logger, filepath.Join(t.ResultPath, string(t.Identifier()), evalmodel.CleanModelNameForFileSystem(t.Model.ID()), t.Language.ID(), repository.Name+".log"))
+	if err != nil {
+		return nil, nil, err
+	}
+	defer logClose()
+
+	log.Printf("Evaluating model %q on task %q using language %q and repository %q", t.Model.ID(), t.Identifier(), t.Language.ID(), repository.Name)
+	defer func() {
+		log.Printf("Evaluated model %q on task %q using language %q and repository %q: encountered %d problems: %+v", t.Model.ID(), t.Identifier(), t.Language.ID(), repository.Name, len(problems), problems)
+	}()
+
+	filePaths, err := t.Language.Files(log, repository.DataPath)
+	if err != nil {
+		return nil, problems, pkgerrors.WithStack(err)
+	}
+
+	var sourceFilesPath []string
+	for _, filePath := range filePaths {
+		if strings.Contains(filePath, t.Language.DefaultTestFileSuffix()) {
+			continue
+		}
+		if !strings.Contains(filePath, t.Language.DefaultFileExtension()) {
+			continue
+		}
+
+		sourceFilesPath = append(sourceFilesPath, filePath)
+	}
+
+	if len(sourceFilesPath) != 1 {
+		return nil, problems, pkgerrors.Errorf("the repository %q must have only one source file, but found several %#v", repository.DataPath, sourceFilesPath)
+	}
+
+	mistakes, err := t.Language.Mistakes(t.Logger, repository.DataPath)
+	if err != nil {
+		return nil, problems, pkgerrors.WithStack(err)
+	}
+	if len(mistakes) == 0 {
+		return nil, problems, pkgerrors.Errorf("the following repository must contain source files with compilation errors: %q", repository.DataPath)
+	}
+
+	repositoryAssessment = metrics.NewAssessments()
+
+	if err := repository.Reset(t.Logger); err != nil {
+		t.Logger.Panicf("ERROR: unable to reset temporary repository path: %s", err)
+	}
```

Comment:
Please reset the repository in the very beginning (before getting the mistakes).